plone/app/discussion/conversation.py

```diff
@@ -89,8 +89,22 @@ def enabled(self):
 
     @property
     def total_comments(self):
+        return self.public_comments()
+
+    def public_comments(self):
+        """Count total comments.
+
+        The @property decorator used on ``total_comments`` method doesn't work
+        when combined with acquisition, that's a fact of life.
+
+        Thus the total_comments indexer (on p.a.discussion.catalog.py) fails
+        to index properly if acquisition is involved.
+
+        To solve it, this method takes over the indexing and is being used by
```

Comment:
@gforcada Why don't we just replace the total_comments implementation? What kind of bbb scenario do you have in mind?